scripts/latency-tracking/app.py

```diff
@@ -0,0 +1,271 @@
+import json
+import os
+import shutil
+import sys
+import time
+import timeit
+from typing import Dict
+from pathlib import Path
+
+# this line is needed here for measuring import time accurately for 1M imports
+import_time = timeit.timeit(stmt='import jina', number=1000000)
+
+from pkg_resources import resource_filename
+from packaging import version
+from jina.types.arrays.memmap import DocumentArrayMemmap
+from jina.helloworld.fashion.helper import (
+    index_generator, download_data, query_generator)
+from jina.parsers.helloworld import set_hw_parser
+from jina.logging.logger import JinaLogger
+from jina import Document, Flow, __version__
+
+try:
+    from jina.helloworld.fashion.executors import MyEncoder, MyIndexer
+except:
+    from jina.helloworld.fashion.my_executors import MyEncoder, MyIndexer
+
+
+# declare base logger
+log = JinaLogger(__name__, "custom_logger")
+
+
+def __doc_generator():
+    # Document generator
+    for i in range(1000):
+        yield Document(text=f'This is the document number: {i}', )
+
+
+def _benchmark_import_time() -> Dict[str, float]:
+    """Benchmark Jina Core import time for 1M imports.
+
+    Returns:
+        A dict mapping of import time in seconds as float number.
+
+    TODO: Figure out How we can measure the import time within a function.
+    """
+    return {
+        'import_time': float(import_time)
+    }
+
+
+def _benchmark_avg_flow_time() -> Dict[str, float]:
+    """Benchmark on a simple flow operation.
+
+    Reurns:
+        A dict mapping of import time in seconds as float number.
+    """
+    fs = [
+        Flow().add(),
+        Flow().add().add(),
+        Flow().add().add().add(),
+        Flow().add().add().add(needs='gateway')
+    ]
+    log.info("Benchmarking average flow time")
+    st = time.perf_counter()
+    for f in fs:
+        with f:
+            f.post(on='/', inputs=__doc_generator, request_size=10)
+    flow_time = time.perf_counter() - st
+    avg_flow_time = flow_time / len(fs)
+    log.info("Average flow time: %f seconds", avg_flow_time)
+
+    return {
+        'avg_flow_time': avg_flow_time
+    }
+
+
+def _benchmark_dam_extend_qps() -> Dict[str, float]:
+    """Benchmark on adding 1M documents to DocumentArrayMemmap.
+
+    Returns:
+        A dict mapping of dam extend time in seconds as float number.
+    """
+    dlist = []
+    dam_size = 1000000
+    dam = DocumentArrayMemmap(os.path.join(os.getcwd(), 'MyMemMap'))
+
+    for i in range(dam_size):
+        dlist.append(Document(text=f'This is the document number: {i}', ))
+
+    log.info("Benchmarking DAM extend")
```

Comment:
Please let's use `single quotes`

Reply:
Without double quotes, the script won't pass `black` check.